src/Tutor.Core/DomainModel/KnowledgeComponents/InteractedWithAssessmentEvent.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Tutor.Core.BuildingBlocks.EventSourcing;
+using Tutor.Core.DomainModel.AssessmentEvents;
+
+namespace Tutor.Core.DomainModel.KnowledgeComponents
+{
+    public class InteractedWithAssessmentEvent : DomainEvent
+    {
+        public int LearnerId { get; set; }
+        public int AssessmentEventId { get; set; }
+        public AssessmentEventInteraction Interaction { get; set; }
+    }
```

Comment:
One thing I dislike about the way I designed this for now is the way that the events look when recorded. 

For example, this is how the event for viewing challenge hints looks for now:
<code>
{"type":"InteractedWithAssessmentEvent","learnerId":1,"assessmentEventId":-5,"interaction": 
{"type":"HintsSought"},"timestamp":"2022-03-16T11:15:21.1638662Z"}
</code>

And this is how the event for viewing challenge solutions looks:
<code>{"type":"InteractedWithAssessmentEvent","learnerId":1,"assessmentEventId":-5,"interaction":{"type":"SolutionSought"},"timestamp":"2022-03-16T11:15:22.4721277Z"}</code>

I feel like it might have been better to make the AssessmentEventInteraction class an event, if we did that the hierarchy of events for AEs would look something like:

![aeevents](https://user-images.githubusercontent.com/57844740/158584092-e1ef7ee3-b146-4944-a921-485d97abd3ab.png)

And if we were adding a functionality that happens when, for example, the SoughtChallengeSolution event does, the event would be moved in the hierarchy to inherit AssessmentEventEvent (better name pending) instead.